packages/tools/guiEditor/src/diagram/workbench.tsx

```diff
@@ -829,7 +830,11 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
         const light = new HemisphericLight("light1", Axis.Y, this._scene);
         light.intensity = 0.9;
 
-        this._guiSize = this._defaultGUISize;
+        if (fromPG) {
+            this.props.globalState.fromPG = true;
+            this._guiSize.width = Number(window.localStorage.getItem("width"));
+            this._guiSize.height = Number(window.localStorage.getItem("height"));

```

Comment:
where were those defined? and what happens is they are not defined?

Reply:
guiSize is initially defined on line 89! It's set to the default size of 1024x1024

Reply:
Is it possible to use the data storage instead of directly the localStorage? or does it make no sense here?

Reply:
Oh I see what you're saying! I think I can directly just use dataStorage. I'll change that now

Reply:
Actually, I just tried using DataStorage and it doesn't seem to be saving the data on reload. I'm not sure why because I see the DataStorage implementation does use localStorage, so I'll try to debug this a bit